pkg/controller_workload/workload/workload_event.go

```diff
@@ -529,15 +529,19 @@ func (svc *ServiceEvent) handleAuthorizationTypeResponse(rsp *service_discovery_
 	for _, resource := range rsp.GetResources() {
 		if err := anypb.UnmarshalTo(resource.Resource, auth, proto.UnmarshalOptions{}); err != nil {
 			log.Errorf("unmarshal failed, err: %v", err)
-			continue
+			return err
```

Comment:
why do we ignore all the workload because of a error., Though i think in theory this could not happen

Reply:
done